test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.netcore31.cs

```diff
@@ -402,6 +421,48 @@ public async Task HttpClientInstrumentationContextPropagation()
             Assert.Equal("b1=v1", baggages.Single());
         }
 
+        [Fact]
+        public async Task HttpClientRedirectTest()
+        {
+            var processor = new Mock<BaseProcessor<Activity>>();
+            using (Sdk.CreateTracerProviderBuilder()
+                       .AddHttpClientInstrumentation()
+                       .AddProcessor(processor.Object)
+                       .Build())
+            {
+#if NETCOREAPP3_1
+                // This implementation of HttpClient uses CurlHandler or SocketsHttpHandler (depending on DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER switch)
+                // and both of them handle redirects internally without calling DiagnosticsHandler.
+                // So, in order for the instrumentation to work correctly in .NET Core 3.1, redirects should be handled manually.
+                using var c = new HttpClient(new HttpClientHandler { AllowAutoRedirect = false });
+                var requestUri = $"{this.url}redirect";
+                while (true)
+                {
+                    var response = await c.GetAsync(requestUri);
+                    if (response.StatusCode == HttpStatusCode.Redirect)
+                    {
+                        requestUri = $"{this.url}{response.Headers.Location}";
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+#else
+                using var c = new HttpClient(new SocketsHttpHandler());
```

Comment:
Is this needed? `new SocketsHttpHandler()`

Reply:
Not really - removed.